<a href="https://colab.research.google.com/github/belkisKOS/data-preprocessing-security-demo/blob/main/01_data_load_and_init.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')


Mounted at /content/drive


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

print("pandas version:", pd.__version__)
print("matplotlib:", plt)
print("seaborn:", sns)


pandas version: 2.2.2
matplotlib: <module 'matplotlib.pyplot' from '/usr/local/lib/python3.11/dist-packages/matplotlib/pyplot.py'>
seaborn: <module 'seaborn' from '/usr/local/lib/python3.11/dist-packages/seaborn/__init__.py'>


In [ ]:
!ls /content/drive/MyDrive


 Atelier		       processed_dataset2
 Atelier2025		       split_balanced_dataset
 Atelier2025_SCI_1st_Dataset   split_normal_withoutaug
 balanced		       test.csv
 best_newest2_model.pth       'Thesis presentation (2).pptx'
'Colab Notebooks'	       titanic_dataset
 final_metrics.csv	       train.csv
 improved_cnn_model.py	       training_newest2_metrics_backup.csv
 normal_withoutaug	       training_newest2_metrics.csv
'normal_withoutaug (1)'       'Untitled document.gdoc'
 original		       val.csv


In [ ]:
!ls /content/drive/MyDrive/titanic_dataset


train_and_test2.csv


In [ ]:
import pandas as pd

# Chemin vers ton fichier sur Drive
path = '/content/drive/MyDrive/titanic_dataset/train_and_test2.csv'
df = pd.read_csv(path)

# Affiche les 5 premières lignes pour vérifier
df.head()


Passengerid   Age     Fare  Sex  sibsp  zero  zero.1  zero.2  zero.3  \
0            1  22.0   7.2500    0      1     0       0       0       0   
1            2  38.0  71.2833    1      1     0       0       0       0   
2            3  26.0   7.9250    1      0     0       0       0       0   
3            4  35.0  53.1000    1      1     0       0       0       0   
4            5  35.0   8.0500    0      0     0       0       0       0   

   zero.4  ...  zero.12  zero.13  zero.14  Pclass  zero.15  zero.16  Embarked  \
0       0  ...        0        0        0       3        0        0       2.0   
1       0  ...        0        0        0       1        0        0       0.0   
2       0  ...        0        0        0       3        0        0       2.0   
3       0  ...        0        0        0       1        0        0       2.0   
4       0  ...        0        0        0       3        0        0       2.0   

   zero.17  zero.18  2urvived  
0        0        0         0  
1        0        0         1  
2        0        0         1  
3        0        0         1  
4        0        0         0  

[5 rows x 28 columns]

In [ ]:
# Aperçu des infos et statistiques de base
print(df.info())
print(df.describe())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 28 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Passengerid  1309 non-null   int64  
 1   Age          1309 non-null   float64
 2   Fare         1309 non-null   float64
 3   Sex          1309 non-null   int64  
 4   sibsp        1309 non-null   int64  
 5   zero         1309 non-null   int64  
 6   zero.1       1309 non-null   int64  
 7   zero.2       1309 non-null   int64  
 8   zero.3       1309 non-null   int64  
 9   zero.4       1309 non-null   int64  
 10  zero.5       1309 non-null   int64  
 11  zero.6       1309 non-null   int64  
 12  Parch        1309 non-null   int64  
 13  zero.7       1309 non-null   int64  
 14  zero.8       1309 non-null   int64  
 15  zero.9       1309 non-null   int64  
 16  zero.10      1309 non-null   int64  
 17  zero.11      1309 non-null   int64  
 18  zero.12      1309 non-null   int64  
 19  zero.1

In [ ]:
# Clone ton repo GitHub (OAuth fonctionne en arrière-plan)
!git clone https://github.com/belkisKOS/data-preprocessing-security-demo.git
%cd data-preprocessing-security-demo


Cloning into 'data-preprocessing-security-demo'...
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 9 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)
Receiving objects: 100% (9/9), 7.10 KiB | 7.10 MiB/s, done.
/content/data-preprocessing-security-demo


In [ ]:
%pwd   # doit retourner /content/data-preprocessing-security-demo


'/content/data-preprocessing-security-demo'

In [ ]:
!ls -la


total 48
drwxr-xr-x 3 root root  4096 Jun  7 18:31 .
drwxr-xr-x 1 root root  4096 Jun  7 18:31 ..
-rw-r--r-- 1 root root 30855 Jun  7 18:31 01_data_load_and_init.ipynb
drwxr-xr-x 8 root root  4096 Jun  7 18:31 .git
-rw-r--r-- 1 root root    34 Jun  7 18:31 README.md


In [ ]:
!git add 01_data_load_and_init.ipynb train_and_test2.csv
!git commit -m "Ajout du notebook initial et du dataset train_and_test2.csv"
!git push origin main


fatal: pathspec 'train_and_test2.csv' did not match any files
Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'root@63314b08faac.(none)')
fatal: could not read Username for 'https://github.com': No such device or address


In [ ]:
from getpass import getpass
token = getpass("GitHub Token : ")
!git push https://{token}@github.com/belkisKOS/data-preprocessing-security-demo.git


GitHub Token : ··········
remote: Permission to belkisKOS/data-preprocessing-security-demo.git denied to belkisKOS.
fatal: unable to access 'https://github.com/belkisKOS/data-preprocessing-security-demo.git/': The requested URL returned error: 403


# Jour 2 (2 juin) : Chargement et inspection initiale

Dans cette étape, nous allons :
- Charger le dataset avec `pd.read_csv()`  
- Afficher un aperçu des données (`df.head()`)  
- Examiner la structure et les types (`df.info()`)  
- Consulter les statistiques descriptives (`df.describe()`)  
- Noter nos premières observations dans une cellule Markdown  


In [2]:
import pandas as pd

# (Si ce n'est pas déjà fait)
from google.colab import drive
drive.mount('/content/drive')

# Chargement
path = '/content/drive/MyDrive/titanic_dataset/train_and_test2.csv'
df = pd.read_csv(path)

# Inspection
display(df.head())
print(df.info())
print(df.describe())


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Passengerid   Age     Fare  Sex  sibsp  zero  zero.1  zero.2  zero.3  \
0            1  22.0   7.2500    0      1     0       0       0       0   
1            2  38.0  71.2833    1      1     0       0       0       0   
2            3  26.0   7.9250    1      0     0       0       0       0   
3            4  35.0  53.1000    1      1     0       0       0       0   
4            5  35.0   8.0500    0      0     0       0       0       0   

   zero.4  ...  zero.12  zero.13  zero.14  Pclass  zero.15  zero.16  Embarked  \
0       0  ...        0        0        0       3        0        0       2.0   
1       0  ...        0        0        0       1        0        0       0.0   
2       0  ...        0        0        0       3        0        0       2.0   
3       0  ...        0        0        0       1        0        0       2.0   
4       0  ...        0        0        0       3        0        0       2.0   

   zero.17  zero.18  2urvived  
0        0        0         0  
1        0        0         1  
2        0        0         1  
3        0        0         1  
4        0        0         0  

[5 rows x 28 columns]

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 28 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Passengerid  1309 non-null   int64  
 1   Age          1309 non-null   float64
 2   Fare         1309 non-null   float64
 3   Sex          1309 non-null   int64  
 4   sibsp        1309 non-null   int64  
 5   zero         1309 non-null   int64  
 6   zero.1       1309 non-null   int64  
 7   zero.2       1309 non-null   int64  
 8   zero.3       1309 non-null   int64  
 9   zero.4       1309 non-null   int64  
 10  zero.5       1309 non-null   int64  
 11  zero.6       1309 non-null   int64  
 12  Parch        1309 non-null   int64  
 13  zero.7       1309 non-null   int64  
 14  zero.8       1309 non-null   int64  
 15  zero.9       1309 non-null   int64  
 16  zero.10      1309 non-null   int64  
 17  zero.11      1309 non-null   int64  
 18  zero.12      1309 non-null   int64  
 19  zero.1

## Observations initiales

- **Taille du jeu de données**  
  1309 lignes × 28 colonnes.

- **Types de variables**  
  - 25 colonnes en `int64`  
  - 3 colonnes en `float64`  
  - Exemples :  
    - `Passengerid` (int64) – identifiant unique  
    - `Age` (float64) – âge du passager  
    - `Fare` (float64) – tarif du billet  
    - `Sex` (int64) – encodage binaire du genre  

- **Valeurs manquantes**  
  - Seule la colonne `Embarked` contient 2 valeurs manquantes (1307 non-null sur 1309).  
  - Les autres colonnes sont complètes.

- **Colonnes à variance nulle**  
  - Les colonnes `zero`, `zero.1`, …, `zero.18` ont toutes leurs valeurs à 0 (pour l’instant non informatives).

- **Statistiques descriptives clés**  
  - **Âge** : moyenne ≈ 29,50 ans, min = 0,17, max = 80, médiane = 28  
  - **Tarif (`Fare`)** : moyenne ≈ 33,28, min = 0,00, max = 512,33, médiane = 14,45  
  - **Genre (`Sex`)** : moyenne ≈ 0,356 (encodage 0/1)  
  - **Fratrie/Conjoint (`sibsp`)** : moyenne ≈ 0,499, max = 8  
  - **Classe (`Pclass`)** : moyenne ≈ 2,29, majoritairement en 3ᵉ classe

- **Cible**  
  - `2urvived` (0/1) – taux de survie moyen ≈ 26,13 %


# Jour 3 (3 juin) : Nettoyage des NA et encodage catégoriel

Dans cette étape, nous allons :
- Identifier les colonnes contenant des valeurs manquantes (`df.isna().sum()`)  
- Supprimer (< 5 % de NA) ou imputer (moyenne/médiane/mode) ces NA  
- Encoder 1–2 variables catégorielles (`pd.get_dummies` ou `LabelEncoder`)  
- Documenter chaque choix en détail  


In [3]:
# 1. Compter les NA par colonne
na_counts = df.isna().sum()
print(na_counts[na_counts > 0])


Embarked    2
dtype: int64


In [4]:
# 2. Suppression des lignes où Embarked est manquant
df = df.dropna(subset=['Embarked'])
print("NA restantes :", df['Embarked'].isna().sum(), "sur", len(df), "lignes")


NA restantes : 0 sur 1307 lignes


In [6]:
# One‐hot encoding correct de la colonne 'Embarked'
df = pd.get_dummies(df, columns=['Embarked'], prefix='Em')
df.head()


Passengerid   Age     Fare  Sex  sibsp  zero  zero.1  zero.2  zero.3  \
0            1  22.0   7.2500    0      1     0       0       0       0   
1            2  38.0  71.2833    1      1     0       0       0       0   
2            3  26.0   7.9250    1      0     0       0       0       0   
3            4  35.0  53.1000    1      1     0       0       0       0   
4            5  35.0   8.0500    0      0     0       0       0       0   

   zero.4  ...  zero.14  Pclass  zero.15  zero.16  zero.17  zero.18  2urvived  \
0       0  ...        0       3        0        0        0        0         0   
1       0  ...        0       1        0        0        0        0         1   
2       0  ...        0       3        0        0        0        0         1   
3       0  ...        0       1        0        0        0        0         1   
4       0  ...        0       3        0        0        0        0         0   

   Em_0.0  Em_1.0  Em_2.0  
0   False   False    True  
1    True   False   False  
2   False   False    True  
3   False   False    True  
4   False   False    True  

[5 rows x 30 columns]

In [8]:
# Affiche toutes les colonnes de ton df
print(df.columns.tolist())


['Passengerid', 'Age', 'Fare', 'Sex', 'sibsp', 'zero', 'zero.1', 'zero.2', 'zero.3', 'zero.4', 'zero.5', 'zero.6', 'Parch', 'zero.7', 'zero.8', 'zero.9', 'zero.10', 'zero.11', 'zero.12', 'zero.13', 'zero.14', 'Pclass', 'zero.15', 'zero.16', 'zero.17', 'zero.18', '2urvived', 'Em_0.0', 'Em_1.0', 'Em_2.0']


In [9]:
df = df.rename(columns={
    'Em_0.0': 'Em_C',   # Cherbourg
    'Em_1.0': 'Em_Q',   # Queenstown
    'Em_2.0': 'Em_S'    # Southampton
})
print(df.columns.tolist())


['Passengerid', 'Age', 'Fare', 'Sex', 'sibsp', 'zero', 'zero.1', 'zero.2', 'zero.3', 'zero.4', 'zero.5', 'zero.6', 'Parch', 'zero.7', 'zero.8', 'zero.9', 'zero.10', 'zero.11', 'zero.12', 'zero.13', 'zero.14', 'Pclass', 'zero.15', 'zero.16', 'zero.17', 'zero.18', '2urvived', 'Em_C', 'Em_Q', 'Em_S']


In [10]:
# Suppression de toutes les colonnes inutiles “zero*”
cols_to_drop = [c for c in df.columns if c.startswith('zero')]
df = df.drop(columns=cols_to_drop)
print("Colonnes restantes :", df.columns.tolist())


Colonnes restantes : ['Passengerid', 'Age', 'Fare', 'Sex', 'sibsp', 'Parch', 'Pclass', '2urvived', 'Em_C', 'Em_Q', 'Em_S']


In [11]:
# One-hot encoding de Pclass
df = pd.get_dummies(df, columns=['Pclass'], prefix='Class')
print(df.columns.tolist())


['Passengerid', 'Age', 'Fare', 'Sex', 'sibsp', 'Parch', '2urvived', 'Em_C', 'Em_Q', 'Em_S', 'Class_1', 'Class_2', 'Class_3']


## Ajustements finaux – Jour 3

- **Colonnes “zero” supprimées** : variance nulle, pas d’information.  
- **Renommage des dummies Embarked** :  
  - `Em_0.0` → `Em_C` (Cherbourg)  
  - `Em_1.0` → `Em_Q` (Queenstown)  
  - `Em_2.0` → `Em_S` (Southampton)  
- **Encodage de Pclass** : one-hot → `Class_1`, `Class_2`, `Class_3`.  
- **Sex** déjà encodé en binaire (0/1).
